task.js

```diff
@@ -0,0 +1,193 @@
+// you can see an example of interaction with the order at the end of the file
+
+class Item {
+    constructor(price, calories) {
+        this.price = price;
+        this.calories = calories;
+    }
+
+    calculatePrice() {
+        throw new Error("abstract method");
+    }
+
+    calculateCalories() {
+        throw new Error("abstract method");
+    }
+}
+
+class HamburgerStuffing {
+    constructor(price, calories) {
+        this.price = price;
+        this.calories = calories;
+    }
+
+    getPrice() {
+        return this.price;
+    }
+
+    getCalories() {
+        return this.calories;
+    }
+}
+
+const cheese = new HamburgerStuffing(10, 20);
+const salad = new HamburgerStuffing(20, 5);
+const potato = new HamburgerStuffing(15, 10);
+
+class Hamburger extends Item {
+    constructor(price, calories, stuffing) {
+        super(price, calories);
+
+        this.stuffings = [stuffing];
+    }
+
+    calculatePrice() {
+        return this.price + this.stuffings.map(s => s.getPrice()).reduce((total, price) => total + price);
```

Comment:
instead of recalculating it each time, you can calculate it only in 2 cases - in constructor, and `addStuffing` method

Reply:
and only return result by method `getPrice`